src/main/java/org/folio/service/transactions/batch/BatchTransactionChecks.java

```diff
@@ -0,0 +1,266 @@
+    checkTransactionsToUpdateHaveMetadata(batch.getTransactionsToUpdate());
+    batch.getTransactionsToCreate()
+      .forEach(transaction -> {
+        if (transaction.getTransactionType() == ALLOCATION) {
+          checkAllocation(transaction);
+        } else if (transaction.getTransactionType() == TRANSFER) {
+          checkTransfer(transaction);
+        }
+      });
+  }
+
+  public static void checkBudgetsAreActive(BatchTransactionHolder holder) {
+    holder.getBudgets().forEach(budget -> {
+      if (budget.getBudgetStatus() != ACTIVE) {
+        throw new HttpException(400, String.format("Cannot process transactions because of an inactive budget for fund %s",
+          holder.getFundCodeForBudget(budget)));
+      }
+    });
+  }
+
+  public static Future<Void> checkTransactionsToDelete(Set<String> idsOfTransactionsToDelete,
+      BatchTransactionDAO transactionDAO, DBConn conn) {
+    // There is currently no budget update when a transaction is deleted.
+    // Batch delete should only be used to delete released encumbrances.
+    if (idsOfTransactionsToDelete.isEmpty()) {
+      return succeededFuture();
+    }
+    return transactionDAO.getTransactionsByIds(idsOfTransactionsToDelete.stream().toList(), conn)
+      .map(transactionsToDelete -> {
+        if (transactionsToDelete.size() != idsOfTransactionsToDelete.size()) {
+          throw new HttpException(400, "One or more transaction to delete was not found");
+        }
+        // NOTE: for the following checks we can't throw an exception because it would prevent the encumbrance script
+        // from working, so instead we just log a warning.
+        // Also in the future we could support auto-releasing encumbrances to delete, but this would not work well
+        // when there are duplicate encumbrances (released/unreleased), so we don't do it yet.
+        transactionsToDelete.forEach(tr -> {
+          if (tr.getTransactionType() != ENCUMBRANCE || tr.getEncumbrance().getStatus() != Encumbrance.Status.RELEASED) {
+            logger.warn("A transaction to delete is not a released encumbrance, id={}", tr.getId());
+          }
+        });
+        return transactionsToDelete.stream()
+          .filter(tr -> tr.getTransactionType() == ENCUMBRANCE)
+          .map(Transaction::getId)
+          .toList();
+      })
+      .compose(idsOfEncumbrancesToDelete -> {
+        if (idsOfEncumbrancesToDelete.isEmpty()) {
+          return succeededFuture(emptyList());
+        }
+        CriterionBuilder criterionBuilder = new CriterionBuilder("OR");
+        idsOfEncumbrancesToDelete.forEach(id -> criterionBuilder.withJson("awaitingPayment.encumbranceId", "=", id));
+        return transactionDAO.getTransactionsByCriterion(criterionBuilder.build(), conn);
+      })
+      .map(pendingPayments -> {
+        if (!pendingPayments.isEmpty()) {
+          logger.warn("An invoice is connected to an encumbrance to delete, id={}", pendingPayments.get(0).getId());
```

Comment:
for reviewers: this call is under investigation by @damien-git , can it be removed, because we just write warn log for it